tests/nlu/featurizers/test_lm_featurizer.py

```diff
@@ -184,57 +270,257 @@
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 1], [1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 1, 1, 1, 2, 1]],
+            ),
+            (
+                "gpt2",
+                None,
+                [
+                    "Good evening.",
+                    "hello",
+                    "you're",
+                    "r. n. b.",
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 2], [1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 2, 1, 1, 3, 1]],
+            ),
+            (
+                "xlnet",
+                None,
+                [
+                    "Good evening.",
+                    "hello",
+                    "you're",
+                    "r. n. b.",
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 1], [1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 1, 1, 1, 3, 1]],
+            ),
+            (
+                "distilbert",
+                None,
+                [
+                    "Good evening.",
+                    "you're",
+                    "r. n. b.",
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 1, 1, 1, 4, 1]],
+            ),
+            (
+                "roberta",
+                None,
+                [
+                    "Good evening.",
+                    "hello",
+                    "you're",
+                    "r. n. b.",
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 2], [1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 2, 1, 1, 3, 1]],
+            ),
+            (
+                "bert",
+                "bert-base-uncased",
+                [
+                    "Good evening.",
+                    "you're",
+                    "r. n. b.",
+                    "rock & roll",
+                    "here is the sentence I want embeddings for.",
+                ],
+                [[1, 1], [1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1, 1, 1, 1, 4, 1]],
+            ),
+        ],
+    )
+
+    general_case_args = (
+        "model_name, model_weights, text, expected_number_of_sub_tokens",
+        [
+            ("bert", None, "sentence embeddings", [1, 2]),
+            ("bert", "bert-base-uncased", "sentence embeddings", [1, 4]),
+            ("gpt", None, "sentence embeddings", [1, 2]),
+            ("gpt2", None, "sentence embeddings", [2, 3]),
+            ("xlnet", None, "sentence embeddings", [1, 3]),
+            ("distilbert", None, "sentence embeddings", [1, 4]),
+            ("roberta", None, "sentence embeddings", [2, 3]),
+        ],
+    )
```

Comment:
I know this is ancient code, but how about we merge the `general_case_args` and `edge_case_args` into one argument set? I don't see a huge gain in separating out the two. Moreover, it duplicates the code below. What do you think?

Reply:
Also, then we can just parameterize the class, rather than parameterizing each function separately.

Reply:
Also, then we can just parameterize the class, rather than parameterizing each function separately.

Reply:
Sounds good